interp/src/debugger/new_parser.rs

```diff
@@ -0,0 +1,99 @@
+use super::source::structures::{GroupContents, NewSourceMap};
+use crate::errors::InterpreterResult;
+use pest_consume::{match_nodes, Error, Parser};
+use std::collections::HashMap;
+type ParseResult<T> = std::result::Result<T, Error<Rule>>;
+type Node<'i> = pest_consume::Node<'i, Rule, ()>;
+
+// include the grammar file so that Cargo knows to rebuild this file on grammar changes
+const _GRAMMAR: &str = include_str!("new_metadata.pest");
+
+#[derive(Parser)]
+#[grammar = "debugger/new_metadata.pest"]
+pub struct MetadataParser;
+
+#[pest_consume::parser]
+impl MetadataParser {
+    fn EOI(_input: Node) -> ParseResult<()> {
+        Ok(())
+    }
+    pub fn num(input: Node) -> ParseResult<u64> {
+        input
+            .as_str()
+            .parse::<u64>()
+            .map_err(|_| input.error("Expected number"))
+    }
+    fn group_name(input: Node) -> ParseResult<String> {
+        input
+            .as_str()
+            .parse::<String>()
+            .map_err(|_| input.error("Expected character"))
+    }
```

Comment:
`&str` means it's a string slice, which can be easily turned into a `String`
with the `to_string` method which is infallible. You can then wrap the whole
thing in an `Ok(_)` to get the desired result
```suggestion
            .to_string()
```